packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import uniqueid from 'lodash.uniqueid';
+import classNames from 'classnames/bind';
+import * as KeyCode from 'keycode-js';
+import styles from './Switch.module.scss';
+
+const cx = classNames.bind(styles);
+
+const SWITCH_STATE = Object.freeze({
+  ON: 'On',
+  OFF: 'Off',
+});
+
+const propTypes = {
+  /**
+    * Whether or not the Switch is enabled ("ON").
+    */
+  isOn: PropTypes.bool,
+
+  /**
+    * If true, the user won't be able to toggle the switch.
+    */
+  isDisabled: PropTypes.bool,
+
+  /**
+    * Text of the label.
+    */
+  labelText: PropTypes.node.isRequired,
+
+  /**
+    * Callback fired when the state is changed. Parameters are (event).
+    */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  isOn: false,
+  isDisabled: false,
+  onChange: undefined,
+};
+
+const Switch = ({
+  isOn,
+  isDisabled,
+  onChange,
+  labelText,
+  ...customProps
+}) => {
+  const [focused, setFocus] = useState(false);
+  const sliderButtonRef = useRef(null);
+
+  const handleClick = (event) => {
+    setFocus(true);
+    sliderButtonRef.current.focus();
+    onChange(event);
+  };
+
+  const handleOnKeyDown = (event) => {
+    if (event.keyCode === KeyCode.KEY_SPACE || event.keyCode === KeyCode.KEY_RETURN) {
+      event.preventDefault();
+      setFocus(true);
+    }
+  };
+
+  const handleOnBlur = () => {
+    setFocus(false);
+  };
+
+  const handleKeyUp = (event) => {
+    if (event.keyCode === KeyCode.KEY_TAB) {
+      setFocus(true);
+    } else if (event.keyCode === KeyCode.KEY_SPACE || event.keyCode === KeyCode.KEY_RETURN) {
+      setFocus(true);
+      onChange(event);
+    }
+  };
```

Comment:
As we transition more and more into hooks we should evaluate the different methods provided. A little research would be necessary to make a decision, but I think in general we should consider using the useCallback hook:

https://reactjs.org/docs/hooks-reference.html#usecallback

Otherwise each re-render all of these callback functions will be reinitialized.

Reply:
Thank you that was good information. maybe we can consider adding useCallback hooks when we do state based updated. for current implementation I think useCallback hooks will not make impact since we are not doing state-based render in switch component.

Reply:
The useCallback hook is for memorizing functions during re-renders and is not restricted to components that have internal state. This component will re-render and initialize those functions every time the consuming application triggers a state/prop update for the checked state.